Add tests and docs for new CodecConfig options (#3477)

987Nabil opened the pull request:
fixes #3477
/claim #3477

987Nabil committed:
Add tests and docs for new CodecConfig options (#3477)

kyri-petrou reviewed (commented):
I might be missing something but the PR doesn't seem to be aligned with / fixing the linked issue.
Was there a change in the decision to integrate it directly with the EndpointAPI and simply apply the config to routes instead?

987Nabil commented:
@kyri-petrou the `CodecConfig` is a Endpoint API concept. It has no effect on the features of the low level API.
I added already the config parameters in #3341 
I could have made it one PR and claim two bounties, but to claim automatically I just added tests and docs here.
